src/design-system/components/menu-item/MenuItem.tsx

```diff
@@ -0,0 +1,107 @@
+import { FC, useRef } from 'react'
+import { MenuItem as MuiMenuItem } from '@mui/material'
+import { ArrowDropDown, ArrowDropUp, CloseRounded } from '@mui/icons-material'
+
+import { MenuItemProps as CommonMenuItemProps } from './MenuItem.types'
+import { MenuItemColorVariant, MenuItemVariant } from './MenuItem.constants'
+import { cn } from '~/utils/cn'
+
+import '~scss-components/menu-item/MenuItem.scss'
+
+interface MenuItemProps extends CommonMenuItemProps {
+  isDropdown?: boolean
+  density?: 1 | 2
+  isToggled?: boolean
+  onRemove?: () => void
+  variant?: MenuItemVariant
+}
+
+const MenuItem: FC<MenuItemProps> = ({
+  title,
+  additionalInfo,
+  alignVariant = 'left',
+  colorVariant = MenuItemColorVariant.Default,
+  density = 1,
+  graphics,
+  isDropdown = false,
+  isToggled = false,
+  isBottomBorder = false,
+  isDisabled = false,
+  onClick,
+  onRemove,
+  variant = MenuItemVariant.Default
+}) => {
+  const graphicsRef = useRef<HTMLButtonElement | null>(null)
+
+  const handleClick = () => {
+    if (graphicsRef.current) {
+      const interactiveElement =
+        graphicsRef.current.querySelector<HTMLInputElement>(
+          'input[type="radio"], input[type="checkbox"], [role="radio"], [role="checkbox"]'
+        )
+
+      if (interactiveElement) {
+        interactiveElement.click()
+        return
+      }
+    }
```

Comment:
Can we get same result with using the <label /> element? Its default behaviour that it is clicking on input that currently inside the label.

Reply:
If you mean like that:
```
const handleClick = () => {
    if (graphicsRef.current) {
      graphicsRef.current.click()
      return
    }

    onClick?.()
  }
```
- unfortunately, checkbox then is not marked as selected on click.
 When I am changing the button tag to label here, I can't get rid of onClick triggering two times, even with event.stopPropagation()